## Ejercicios 5.18, 5.19 y 5.22 
### libro Carter Hills et al (2017)

## Aldo Mendoza Rebollar 201930811

In [3]:

install.packages("knitr", type = "binary")


  There is a binary version available (and will be installed) but the
  source version is later:
      binary source
knitr   1.33   1.42



also installing the dependency 'stringr'



package 'stringr' successfully unpacked and MD5 sums checked
package 'knitr' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\aldom\AppData\Local\Temp\Rtmp8yXYbv\downloaded_packages


In [5]:
install.packages("printr", type = "binary")


  There is a binary version available (and will be installed) but the
  source version is later:
       binary source
printr  0.1.1    0.3

package 'printr' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\aldom\AppData\Local\Temp\Rtmp8yXYbv\downloaded_packages


In [8]:
install.packages("rmarkdown",type = "binary")


  There is a binary version available (and will be installed) but the
  source version is later:
          binary source
rmarkdown    2.8   2.20



also installing the dependency 'tinytex'



package 'tinytex' successfully unpacked and MD5 sums checked
package 'rmarkdown' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\aldom\AppData\Local\Temp\Rtmp8yXYbv\downloaded_packages


In [11]:
install.packages("ggplot2",type="binary")


  There is a binary version available (and will be installed) but the
  source version is later:
        binary source
ggplot2  3.3.3  3.4.1



also installing the dependencies 'lifecycle', 'scales'



package 'lifecycle' successfully unpacked and MD5 sums checked
package 'scales' successfully unpacked and MD5 sums checked
package 'ggplot2' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\aldom\AppData\Local\Temp\Rtmp8yXYbv\downloaded_packages


In [16]:
install.packages("broom",type="binary")



  There is a binary version available (and will be installed) but the
  source version is later:
      binary source
broom  0.7.6  1.0.3



also installing the dependencies 'tidyselect', 'dplyr', 'generics', 'tidyr'



package 'tidyselect' successfully unpacked and MD5 sums checked
package 'dplyr' successfully unpacked and MD5 sums checked
package 'generics' successfully unpacked and MD5 sums checked
package 'tidyr' successfully unpacked and MD5 sums checked
package 'broom' successfully unpacked and MD5 sums checked

The downloaded binary packages are in
	C:\Users\aldom\AppData\Local\Temp\Rtmp8yXYbv\downloaded_packages


In [17]:
library("knitr")
library("xtable")
library("printr")
library("stargazer")
library("rmarkdown")
library("openxlsx")
library("readxl")
library("ggplot2")
library(broom)

Warning message:
"package 'broom' was built under R version 3.6.3"

## 5.18

Consider the following two expenditure share equations where the budget share for food WFOOD, and the budget share for clothing WCLOTH, are expressed as functions of total expenditure TOTEXP.

$WFOOD= \beta_1 + \beta_2 ln(TOTEXP)+ e_F$

$WCLOTH= \alpha_1 + \alpha_2 ln(TOTEXP)+ e_F$

**a) A commodity is regarded as a luxury if the coefficient of $\ln(TOTEXP)$ is positive and a necessity if it is negative. What signs would you expect for $\beta_2$ and $\alpha_2$?**

Los bienes son un artículo de lujo, si solo si, el coeficiente de ln(TOTEXP) es positivo y por lo tanto, una necesidad si es negativo. Siendo comida es un gasto necesario, por lo que esperamos que  $\beta_2$ sea negativo.

**b) Using the data in the file london5, estimate the above equations using observations on households with one child. Comment on the estimates and their significance. Can you explain any possibly counterintuitive outcomes?**

In [30]:
london5 <- read_excel("D:/8 semestre/ECONOMETRIA/Bases de datos/london5.xlsx")
attach(london5)
head(london5)

The following objects are masked from london5 (pos = 3):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 4):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 5):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 6):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 7):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 8):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 9):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 10):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 11):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 12):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 13):

wfood  wcloth totexp age nk
1 0.4272 0.0000  50    25  2 
2 0.3739 0.0091  90    39  2 
3 0.1941 0.0012 180    47  2 
4 0.4438 0.0539  80    33  2 
5 0.3331 0.0399  90    31  1 
6 0.3752 0.1170  70    24  1

In [32]:
attach(london5)
child_1<-london5[london5$nk==1,]
child_2<-london5[london5$nk==2,]
mod1<-lm(wfood~log(totexp),data=child_1)
B1<-coef(mod1)[[1]]
B2<-coef(mod1)[[2]]
smod1 <- summary(mod1)
smod1$coefficients

The following objects are masked from london5 (pos = 3):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 4):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 5):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 6):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 7):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 8):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 9):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 10):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 11):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 12):

    age, nk, totexp, wcloth, wfood

The following objects are masked from london5 (pos = 13):

Estimate   Std. Error t value   Pr(>|t|)    
(Intercept)  1.0754855 0.05313181  20.24184 5.096274e-60
log(totexp) -0.1635929 0.01196725 -13.67005 4.284285e-34

El valor esperado de la variable alimentos para un gasto total y las demas variables nulas, se espera la necesidad de 1.0754855. 
Por otro lado, un cambio de una unidad en el ln(TOTEXP) da como resultado una disminución de 0.1635929 en el presupuesto de comida.

In [4]:
mod2<-lm(wcloth~log(totexp),data=child_1)
B_1<-coef(mod2)[[1]]
B_2<-coef(mod2)[[2]]
smod2 <- summary(mod2)
smod2$coefficients

Estimate    Std. Error t value   Pr(>|t|)    
(Intercept) -0.18853178 0.05258504 -3.585274 3.869459e-04
log(totexp)  0.06492601 0.01184410  5.481717 8.317337e-08

Si tuvieramos un gasto de cero, se estima que las necesidades básicas de ropa son de menos 0.1885318. Además, indica que un cambio de una unidad en el ln(TOTEXP) da como resultado un cambio de 0.06492601 en el presupuesto de ropa. Confirmando que la ropa por el signo es mas un lujo que una necesidad básica.

**c)  Using a 1% significance level, test $H_0 ∶β2 \geq 0$ against the alternative $H_1 ∶\beta_2 < 0$. Why might you set up the hypotheses in this way?**

El planteamiento de la pregunta esta sesgada al buscar rechazar $H_0:β2 \geq 0$, ya que caemos en error falso-positivo.
De tal manera que, $H_0$ no se debe aceptar.

In [5]:
alpha<-0.01 
df<-mod1$df.residual
tcr<- qt(1-alpha,df)
m1seb2<-sqrt(vcov(mod1)[2,2]) 
t<-B2/m1seb2
tcr 
t

[1] 2.337564

[1] -13.67005

Es una contradicción la prueba de hipótesis del supuesto inicial, ya que t no está en la región de rechazo podemos aceptar $H_0$.  Lo cual nos indicaría que la comida es un bien de lujo, siendo una contradicción.

**d) Using a 1% significance level, test $H_0 ∶\alpha_2 \geq 0$ against the alternative $H_1 ∶\alpha2 < 0$. Why might you set up the hypotheses in this way?**

De igual manera nos contradecimos al utilizar esta hipótesis, ya que:


In [6]:
alpha<-0.01
df<-mod2$df.residual
tcr<- qt(1-alpha,df)
m2sea2<-sqrt(vcov(mod2)[2,2])
t<-B_2/m2sea2
tcr
t

[1] 2.337564

[1] 5.481717

Nos contradecimos del supuesto inicial, ya que la prueba de hipótesis nos indicarían rechazar $H_0$ lo cual nos indicaria que la ropa es un bien de lujo a favor de $H_1$,  que pueda indicar un fallo en el modelo.

**e) Estimate the two equations using observations on households with two children. Construct 95% interval estimates for $\beta_2$ and $\alpha_2$ for both one- and two-child households. Based on these interval estimates, would you conjecture that the coefficients of $\ln(TOTEXP)$ are the same or different for one- and two-child households.**

# Ecuaciones para uno y dos niños

In [7]:
mod3<-lm(wfood~log(totexp),data=child_2)
B1_2<-coef(mod3)[[1]]
B2_2<-coef(mod3)[[2]]
smod3 <- summary(mod3)

mod4<-lm(wcloth~log(totexp),data=child_2)
B1_3<-coef(mod4)[[1]]
B2_3<-coef(mod4)[[2]]
smod4 <- summary(mod4)

smod3$coefficients
smod4$coefficients

Estimate   Std. Error t value   Pr(>|t|)    
(Intercept)  0.9557926 0.05090624  18.77555 3.156469e-60
log(totexp) -0.1291262 0.01124566 -11.48231 2.591743e-27

Estimate    Std. Error t value   Pr(>|t|)    
(Intercept) -0.25975501 0.05083714 -5.109552 4.566088e-07
log(totexp)  0.08098074 0.01123040  7.210852 2.009840e-12

Los intervalos de confianza son los siguientes:

In [8]:
intconf_1<-confint(mod1,parm = c("log(totexp)"),level=.95)
intconf_2<-confint(mod3,parm = c("log(totexp)"),level=.95)
TablaConf<-data.frame(rbind(intconf_1,intconf_2))
rownames(TablaConf)<-c("1 child","2 childs")
colnames(TablaConf)<-c('Límite_Inferior', 'Límite_Superior')
TablaConf

Límite_Inferior Límite_Superior
1 child  -0.1871336      -0.1400522     
2 childs -0.1512195      -0.1070329

In [9]:
intconf_3<-confint(mod2,parm = c("log(totexp)"),level=.95)
intconf_4<-confint(mod4,parm = c("log(totexp)"),level=.95)
TablaConf1<-data.frame(rbind(intconf_3,intconf_4))
rownames(TablaConf1)<-c("1 child","2 childs")
colnames(TablaConf1)<-c('Límite_Inferior', 'Límite_Superior')
TablaConf1

Límite_Inferior Límite_Superior
1 child  0.04162758      0.08822445     
2 childs 0.05891740      0.10304407

Aparentemente son diferentes, por lo que el gasto puede similar, pero no es igual,

**f) Use all observations to estimate the following two equations and test, at a 95% significance level, whether your conjectures in part (e) are correct. (NK = number of children in the household.)**

$WFOOD=\gamma_1+ \gamma_2\ln(TOTEXP)+ \gamma_3NK+ \gamma_4NK \times \ln(TOTEXP)+e_F$

$CLOTH=\delta_1+ \delta_2\ln(TOTEXP)+ \delta_3NK+ \delta_4NK \times \ln(TOTEXP)+e_F$

In [10]:
mod5<-lm(wfood~log(totexp)+nk+I(nk*log(totexp)),data=london5)
mod5$coefficients
(G1<-coef(mod5)[[1]])
(G2<-coef(mod5)[[2]])
(G2<-coef(mod5)[[3]])
(G2<-coef(mod5)[[4]])

[1] 1.195178

[1] -0.1980597

[1] -0.1196929

[1] 0.03446676

In [11]:
mod6<-lm(wcloth~log(totexp)+nk+I(nk*log(totexp)),data=london5)
(S1<-coef(mod6)[[1]])
(S2<-coef(mod6)[[2]])
(S2<-coef(mod6)[[3]])
(S2<-coef(mod6)[[4]])

[1] -0.1173086

[1] 0.04887129

[1] -0.07122322

[1] 0.01605472

**g)Compare the estimates for $\frac{\partial E(WFOOD|X)}{\partial ln(TOTEXP)}$ from (XR5.18.1) for NK = 1, 2 with those from (XR5.18.3) for NK = 1, 2.**

In [12]:
Tabla_Comparar<-rbind(cbind(B_2,B2_3),cbind(B2,B2_2))
colnames(Tabla_Comparar)<-c("1 child","2 childs")
rownames(Tabla_Comparar)<-c("WFOOD (XR5.18.1)","WCLOTH (XR5.18.3)")
Tabla_Comparar

1 child     2 childs   
WFOOD (XR5.18.1)   0.06492601  0.08098074
WCLOTH (XR5.18.3) -0.16359293 -0.12912617

## 5.19

Consider the following expenditure share equation where the budget share for food WFOOD is expressed as a function of total expenditure TOTEXP.

$WFOOD=\beta_1 + \beta_2+\ln(TOTEXP)+e_F$

In Exercise 4.12, it was noted that the elasticity of expenditure on food with respect to total expenditure
is given by

$\varepsilon=1+\frac{\beta_2}{\beta_1+\beta_2\ln(TOTEXP)}$

Also, in Exercise 5.18 it was indicated that a good is a necessity if $\beta_2 < 0.$

**a) Show that $\beta_2 < 0$ if and only if $\varepsilon < 1$. That is, a good is a necessity if its expenditure elasticity is less than one (inelastic).**

Suponemos que $\varepsilon<1$ tal que
1 +$\frac{\beta_2}{\beta_1 +\beta_2 \ln(TOTEXP)}<1$
por lo que $\frac{\beta_2}{\beta_1 +\beta_2 \ln(TOTEXP)}<0$

Siendo, $\beta_1 +\beta2 \ln(TOTEXP)$ debe ser positivo
($\beta_2<0$), vemos que si despejamos $\beta_2$ nos quedaria $\beta_2<0$ y $\ln(TOTEXP)>0$ 

Por ende, $\beta_2<0$,entonces $\beta_2\ln(TOTEXP)<0$,entonces $\beta_2\ln(TOTEXP)+1<1$,entonces $\varepsilon=1+\frac{\beta_2}{\beta_1+\beta_2\ln(TOTEXP)}$ ,tal que  $(\beta_1 +\beta_2 \ln(TOTEXP))(\varepsilon-1)<0$ y $(\varepsilon-1)<0$ implica que $\varepsilon<1$

Comprobemos con el valor de la elasticidad

In [13]:
london5 <- read_excel("D:/8 semestre/ECONOMETRIA/Bases de datos/london5.xlsx")
attach(london5)
mod1 <- lm(wfood~log(totexp), data = london5)
smod1 <- summary(mod1)
smod1

The following objects are masked from london5 (pos = 3):

    age, nk, totexp, wcloth, wfood


The following objects are masked from london5 (pos = 4):

    age, nk, totexp, wcloth, wfood





Call:
lm(formula = wfood ~ log(totexp), data = london5)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.34226 -0.05737 -0.00200  0.06005  0.31348 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept)  0.993838   0.037114   26.78   <2e-16 ***
log(totexp) -0.140509   0.008261  -17.01   <2e-16 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.08913 on 848 degrees of freedom
Multiple R-squared:  0.2544,	Adjusted R-squared:  0.2535 
F-statistic: 289.3 on 1 and 848 DF,  p-value: < 2.2e-16


In [14]:
b1<-coef(mod1)[1]
b2<-coef(mod1)[2]
elasticidad<-1+(b2/(b1+b2+log(totexp)))

In [15]:
max(elasticidad)<1

[1] TRUE

**b) Use observations in the data file london5 to estimate (XR5.19.1) and comment on the results.**

Los coeficientes son los siguientes por lo que la ecuación seria:

In [16]:
smod1$coefficients

Estimate   Std. Error  t value   Pr(>|t|)     
(Intercept)  0.9938385 0.037113599  26.77828 5.797769e-115
log(totexp) -0.1405088 0.008261118 -17.00845  4.833770e-56

$\hat{WFOOD}=.993838-.14050\ln(TOTEXP)$

Estimamos que un cambio del aprox 1% el gasto total disminuirá en presupuesto destinada a los alimentos en 0.0014. Y confirmamos que la comida es una necesidad, por el signo del coeficiente.

**c) Find point estimates and 95% interval estimates for the mean budget share for food, for total expenditure values (i) TOTEXP = 50 (the fifth percentile of TOTEXP), (ii) TOTEXP = 90 (the median),
and (iii) TOTEXP = 170 (the 95th percentile).**

In [17]:
TOTEXP = c(50, 90, 170)
Estimacion_puntual=b1+b2*log(TOTEXP)
Estimacion_puntual

[1] 0.4441649 0.3615757 0.2722137

In [18]:
tabla=c()
for(i in TOTEXP){
totexp <- i
newdata <- data.frame(log(totexp))
pred <- predict(mod1, newdata = newdata, interval = "confidence", level = 0.95)
est_df <- c(pred[1],pred[2],pred[3])
tabla=rbind(tabla,est_df)
}
colnames(tabla)=c('Estimación','Límite_Inferior', 'Límite_Superior')
rownames(tabla)=c('TOTEXP=50','TOTEXP=90', 'TOTEXP=170')
tabla

Estimación Límite_Inferior Límite_Superior
TOTEXP=50  0.4441649  0.4332097       0.4551201      
TOTEXP=90  0.3615757  0.3555641       0.3675873      
TOTEXP=170 0.2722137  0.2599657       0.2844616

**d)Find point estimates and 95% interval estimates for the elasticity $\varepsilon$, for total expenditure values
(i) TOTEXP = 50 (the fifth percentile), (ii) TOTEXP = 90 (the median), and (iii) TOTEXP = 170
(the 95th percentile).**

In [19]:
varb1<-vcov(mod1)[1,1]
varb2<-vcov(mod1)[2,2]
covb1b2<-vcov(mod1)[1,2]
alpha<-0.05
df<-mod1$df.residual
tcr<-qt(1-alpha/2,df)
cvar=c()
cest=c()
cLV=c()
cUV=c()
for(i in TOTEXP){
    var=varb1+log(i)**2* varb2 +2*log(i)*covb1b2
    est<-1+(b2/(b1+b2*log(i)))
    cvar=c(cvar,var)
    cest=c(cest,est)
    LV<-est-tcr*sqrt(var) 
    UV<-est+tcr*sqrt(var)
    cLV=c(cLV,LV)
    cUV=c(cUV,UV)
}
tabla2=cbind(cest,cLV,cUV)
colnames(tabla2)=c("Estimación",'Límite_Inferior', 'Límite_Superior')
rownames(tabla2)=c('TOTEXP=50','TOTEXP=90', 'TOTEXP=170')
tabla2

Estimación Límite_Inferior Límite_Superior
TOTEXP=50  0.6836562  0.6727011       0.6946114      
TOTEXP=90  0.6113987  0.6053871       0.6174103      
TOTEXP=170 0.4838291  0.4715811       0.4960770

**e) Comment on how the mean budget share and the expenditure elasticity for food change as total
expenditure changes. How does the reliability of estimation change as total expenditure changes?**


Para los bienes necesarios se encuetra una relacion inversamente proporcional a el gasto total con respecto el presupusto y la elasticidad como en el caso de los alimentos 
La estimación de la proporción del presupuesto es más precisa en el gasto total y menor en los percentiles más altos. La estimación de la elasticidad se vuelve menos confiable a medida que aumenta el gasto total.

## 5.22

Using the data in the file toody5, estimate the model

$Y_t=\beta_1+\beta_2TREND_t+\beta_3RAIN_t+\beta_4RAIN_t^2+\beta_5(RAIN_t \times TREND_t)+ e_t$

where $Y_t$ = wheat yield in tons per hectare in the Toodyay Shire of Western Australia in year t; $TREND_t$ is a trend variable designed to capture technological change, with observations 0, 0.1, 0.2, ..., 4.7; 0 is for the year 1950, 0.1 is for the year 1951, and so on up to 4.7 for the year 1997; $RAIN_t$ is total rainfall in decimeters (dm) from May to October (the growing season) in year t (1 decimeter = 4 inches).

**a)Report your estimates, standard errors, t-values, and p-values in a table.**

In [20]:
toody <- read_excel("/Users/alexandertrejo/Documents/8VO SEMESTRE/Econometria/Bases de datos/toody5.xlsx")
head(toody)
attach(toody)

dateid01   rain  t trend y      year
1 1950-01-01 4.099 1 0.0   1.3508 1950
2 1951-01-01 3.574 2 0.1   0.9476 1951
3 1952-01-01 4.258 3 0.2   1.2569 1952
4 1953-01-01 4.790 4 0.3   0.9142 1953
5 1954-01-01 3.648 5 0.4   1.1022 1954
6 1955-01-01 6.035 6 0.5   1.1897 1955

The following object is masked _by_ .GlobalEnv:

    t




In [21]:
mod1 <- lm(y~trend+rain+I(rain^2)+rain*trend, data=toody)
smod1 <- summary(mod1)
tabla <- data.frame(xtable(smod1))
names(tabla) <- c("Estimate", "Std.Error","t-value", "p-value")
kable(tabla, digits=3, align='c', 
      caption="Summary of the $Toody$ model")



Table: Summary of the $Toody$ model

|            | Estimate | Std.Error | t-value | p-value |
|:-----------|:--------:|:---------:|:-------:|:-------:|
|(Intercept) |  -2.675  |   0.997   | -2.684  |  0.010  |
|trend       |  0.678   |   0.192   |  3.533  |  0.001  |
|rain        |  1.508   |   0.431   |  3.501  |  0.001  |
|I(rain^2)   |  -0.157  |   0.047   | -3.369  |  0.002  |
|trend:rain  |  -0.093  |   0.047   | -1.985  |  0.054  |

**b)For 1974, when TREND = 2.4 and RAIN = 4.576, use a 5% significance level to test the null
hypothesis that extra rainfall will not increase expected yield against the alternative that it will
increase expected yield.**

In [22]:
alpha <- .05
df <- mod1$df.residual
L <- mod1$coeff[3]+2*4.576*mod1$coeff[4]+2.4*mod1$coeff[5]
a <- c(0,0,1, 2*4.576, 2.4)
b <- as.numeric(coef(mod1))
V <- vcov(mod1) 
A <- as.vector(a)
varL <- as.numeric(t(A) %*% V %*% A)
se <- sqrt(varL)
t <- L/se
tcr <- qt(1-alpha,df)
t;tcr

rain 
-2.179325

[1] 1.681071

Rechazamos $H_0$ , ya que t está en la region de rechazo.
Por lo tanto, significa que la lluvia incrementa producira una aumento en el rendimiento esperado.
Para un nivel de significancia del 5%

**c)Assuming rainfall is equal to its median value of 3.8355 dm, find point and 95% interval estimates
of the expected improvement in wheat yield from technological change over the period 1960–1995.**

A nivel de datos si se filtran quedaria lo siguiente

In [23]:
library("dplyr")
toody$dateid01 <-as.Date(toody$dateid01,"%Y/%m/%d")
toody1960_1995= toody %>%
  select(dateid01, rain,t,trend,y,year) %>%
  filter(dateid01 >= "1960-01-01" & dateid01 <= "1995-01-01")
head(toody1960_1995)
tail(toody1960_1995)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union


Warning message in as.POSIXlt.POSIXct(x, tz = tz):
“unknown timezone '%Y/%m/%d'”


dateid01   rain  t  trend y      year
1 1960-01-01 4.245 11 1.0   1.1427 1960
2 1961-01-01 3.198 12 1.1   0.7998 1961
3 1962-01-01 4.735 13 1.2   0.8468 1962
4 1963-01-01 6.097 14 1.3   0.8938 1963
5 1964-01-01 5.724 15 1.4   0.9006 1964
6 1965-01-01 4.645 16 1.5   1.0955 1965

dateid01   rain  t  trend y      year
1 1990-01-01 4.457 41 4.0   2.0983 1990
2 1991-01-01 4.396 42 4.1   1.9555 1991
3 1992-01-01 3.746 43 4.2   1.8192 1992
4 1993-01-01 3.651 44 4.3   2.3881 1993
5 1994-01-01 4.896 45 4.4   2.2106 1994
6 1995-01-01 4.797 46 4.5   2.4475 1995

In [24]:
mod1 <- lm(y~trend+rain+I(rain^2)+rain*trend, data=toody1960_1995)
Estimacion_puntual <- data.frame("trend"=c(1,4.5),
                    "rain"=3.8355,"I(rain^2)"=14.7111,
                    "trend*rain"=c(3.8355, 17.25975))
Estimacion_puntual

trend rain   I.rain.2. trend.rain
1 1.0   3.8355 14.7111    3.83550  
2 4.5   3.8355 14.7111   17.25975

In [25]:
predict(object=mod1, newdata=Estimacion_puntual, interval="confidence", level=0.95)

fit       lwr       upr     
1 0.9069325 0.6862871 1.127578
2 2.2931560 2.1054286 2.480883

In [26]:
predict(object=mod1, newdata=Estimacion_puntual, interval="prediction", level=0.95)

fit       lwr       upr     
1 0.9069325 0.3374234 1.476442
2 2.2931560 1.7355739 2.850738

Para 35 años, quedaria lo siguiente:

In [43]:
mod1 <- lm(y~trend+rain+I(rain^2)+(rain*trend), data=toody)
alpha <- .05
df <- mod1$df.residual
L <- 3.5*mod1$coeff[2]+3.8355*3.5*mod1$coeff[5]
a <- c(0,3.5,0, 0, 3.8355*3.5)
b <- as.numeric(coef(mod1))
V <- vcov(mod1) 
A <- as.vector(a)
varL <- as.numeric(t(A) %*% V %*% A)
se <- sqrt(varL)
t <- L/se
tcr <- qt(1-alpha/2,df)
lowbL <- L-tcr*se
upbL <- L+tcr*se
paste("El intervalo de confianza es :","(",lowbL,upbL,")")

[1] "El intervalo de confianza es : ( 0.889830329177559 1.36431568603994 )"

**d)There is concern that climate change is leading to a decline in rainfall over time. To test this hypoth- esis, estimate the equation $RAIN = \alpha_1 + \alpha_2 TREND + e$. Test, at a 5% significance level, the null hypothesis that mean rainfall is not declining over time against the alternative hypothesis that it is
declining.**

In [28]:
mod2 <- lm(rain~trend, data=toody)
smod2 <- summary(mod2)
tabla <- data.frame(xtable(smod2))
names(tabla) <- c("Estimate", "Std.Error","t-value", "p-value")
kable(tabla, digits=3, align='c', caption="Summary of the $Toody3$ model")



Table: Summary of the $Toody3$ model

|            | Estimate | Std.Error | t-value | p-value |
|:-----------|:--------:|:---------:|:-------:|:-------:|
|(Intercept) |  4.393   |   0.243   | 18.078  |  0.000  |
|trend       |  -0.152  |   0.089   | -1.709  |  0.094  |

**e)Using the estimated equation from part (d), estimate mean rainfall in 1960 and in 1995.**

In [29]:
b1 <- coef(mod2)[1]
b2 <- coef(mod2)[2]
rain1 <- b1+b2*1
rain1

(Intercept) 
   4.240694

In [30]:
b1 <- coef(mod2)[1]
b2 <- coef(mod2)[2]
rain2 <- b1+b2*4.5
rain2

(Intercept) 
   3.707899

**f)Suppose that $TREND_{1995} = TREND_{1960}$, implying there had been no technological change
from 1960 to 1995. Use the estimates from part (e) to find an estimate of the decline in mean yield from 1960 to 1995 attributable to climate change.**

In [31]:
b1 <- coef(mod1)[1]
b2 <- coef(mod1)[2]
b3 <- coef(mod1)[3]
b4 <- coef(mod1)[4]
b5 <- coef(mod1)[5]
y1 <- b1+b2*1+b3*4.245+b4*4.245^2+b5*1*4.245
y1

(Intercept) 
  0.9573535

In [32]:
b1 <- coef(mod1)[1]
b2 <- coef(mod1)[2]
b3 <- coef(mod1)[3]
b4 <- coef(mod1)[4]
b5 <- coef(mod1)[5]
y2 <- b1+b2*1+b3*4.797+b4*4.797^2+b5*1*4.797
y2

(Intercept) 
  0.9505572

In [33]:
Estimacion_puntual_1 <- data.frame("trend"=1,
                    "rain"=c(4.245, 4.797),
                    "I(rain^2)"=c(18.02, 23.01121),
                    "trend*rain"=c(4.245, 4.797))
Estimacion_puntual_1

trend rain  I.rain.2. trend.rain
1 1     4.245 18.02000  4.245     
2 1     4.797 23.01121  4.797

In [34]:
predict(object=mod1, newdata=Estimacion_puntual_1, interval="prediction", level=0.95)

fit       lwr       upr     
1 0.9573535 0.3923374 1.522370
2 0.9505572 0.3857573 1.515357

In [35]:
b1 <- coef(mod1)[1]
b2 <- coef(mod1)[2]
b3 <- coef(mod1)[3]
b4 <- coef(mod1)[4]
b5 <- coef(mod1)[5]
y3 <- b1+b2*1+b3*mean(rain)+b4*mean(rain)^2+b5*1*mean(rain)
y3

(Intercept) 
  0.9374214

**g)Suppose that $E RAIN_{1995} = E RAIN_{1960} $, implying there had been no rainfall change from 1960 to 1995. Find an estimate of the increase in mean yield from 1960 to 1995 attributable to technological change.**

In [36]:
b1 <- coef(mod1)[1]
b2 <- coef(mod1)[2]
b3 <- coef(mod1)[3]
b4 <- coef(mod1)[4]
b5 <- coef(mod1)[5]
y4 <- b1+b2*4.5+b3*mean(rain)+b4*mean(rain)^2+b5*4.5*mean(rain)
y4

(Intercept) 
   2.281116

**g)Suppose that $E RAIN_{1995} = E RAIN_{1960} $, implying there had been no rainfall change from 1960 to 1995. Find an estimate of the increase in mean yield from 1960 to 1995 attributable to technological change.**

In [37]:
Estimacion_puntual_3 <- data.frame("trend"=c(1,4.5),
                    "rain"=mean(rain),"I(rain^2)"=mean(rain)^2,
                    "trend*rain"=c(4.240694, 19.083123))
Estimacion_puntual_3

trend rain     I.rain.2. trend.rain
1 1.0   4.035188 16.28274   4.240694 
2 4.5   4.035188 16.28274  19.083123

In [38]:
predict(object=mod1, newdata=Estimacion_puntual_3, interval="confidence", level=0.95)

fit       lwr      upr     
1 0.9374214 0.723712 1.151131
2 2.2811160 2.104082 2.458150

**h) Compare the estimates you obtained in parts (c), (f), and (g).**

In [39]:
comparacion=rbind(predict(object=mod1, newdata=Estimacion_puntual, interval="confidence", level=0.95),
predict(object=mod1, newdata=Estimacion_puntual, interval="prediction", level=0.95),
predict(object=mod1, newdata=Estimacion_puntual_1, interval="prediction", level=0.95),
predict(object=mod1, newdata=Estimacion_puntual_3, interval="confidence", level=0.95))

rownames(comparacion)=c("RAIN_3.8355_1","RAIN_3.8355_2","Ip_1","Ip_2",
                        "cambio_climatico_1","cambio_climatico_2",
                        "cambio_tecnologia_1","cambio_tecnologia_2")
comparacion

fit       lwr       upr     
RAIN_3.8355_1       0.9069325 0.6862871 1.127578
RAIN_3.8355_2       2.2931560 2.1054286 2.480883
Ip_1                0.9069325 0.3374234 1.476442
Ip_2                2.2931560 1.7355739 2.850738
cambio_climatico_1  0.9573535 0.3923374 1.522370
cambio_climatico_2  0.9505572 0.3857573 1.515357
cambio_tecnologia_1 0.9374214 0.7237120 1.151131
cambio_tecnologia_2 2.2811160 2.1040820 2.458150

Comparando los intervalos, concluimos que la tecnologia afecta de manera más poderada que el cmbio climatico.